File: inttostr.py
def inttostr(num):
	convertString = "0123456789"
	digits = ""
	while(num):
		digit = num % 10
		digits = convertString[digit] + digits
		num = num//10
	return digits

def inttostrRec(num,base):
	convertString = "0123456789ABCDEF"
	if num < base: return convertString[num]
	return inttostrRec(num//base,base) + convertString[num % base]

test = 123
print(inttostrRec(test,16))

File: c/inttostr.c
#include<stdio.h>
#include<math.h>

void inttostr(int num){
	int strMap[10] = {'0','1','2','3','4','5','6','7','8','9'};
	int digit;
	int lengthString = ceil(log(num));
	char result[lengthString];
	int i =0;
	while(num){
		digit = num % 10;
		num = floor(num/10);
		result[i] = strMap[digit];
		i++;
	}
	for(i;i>0;i--) printf("%c",result[i-1]);
}

int main(){
	int test = 123;
	inttostr(test);
	return 1;
}
